server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -660,7 +661,17 @@ public SearchExecutionContext newSearchExecutionContext(
             shardRequestIndex,
             indexSettings,
             indexCache.bitsetFilterCache(),
-            indexFieldData::getForField,
+            new SearchExecutionContext.IndexFieldDataLookup() {
```

Comment:
Here is where the single interface with both lookup and predicate methods is "worse": the anonymous class is not brilliant to read.
If we like the single interface, I would like to extract a static inner class (adapter) to make it nicer (more readable). Thoughts?